source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include "envoy/config/filter/udp/udp_proxy/v2alpha/udp_proxy.pb.h"
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/network/filter.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/network/utility.h"
+
+#include "absl/container/flat_hash_set.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace UdpProxy {
+
+class UdpProxyFilterConfig {
+public:
+  UdpProxyFilterConfig(Upstream::ClusterManager& cluster_manager, TimeSource& time_source,
+                       const envoy::config::filter::udp::udp_proxy::v2alpha::UdpProxyConfig& config)
```

Comment:
Do we really want to pass around envoy::config::filter::udp::udp_proxy::v2alpha::UdpProxyConfig? Keeping config_.cluster() in this class seems sufficient to me.

Reply:
There will be more settings later. I may change this but I don't think it matters and I can do this later if needed?

Reply:
SGTM